### PR TITLE
Make ``jax_tracer`` handle ``__name__`` for ``functools.partial``

### PR DESCRIPTION
**Pull request by tzunghanjuang**
**Context:** 
Add if-else case to get ``__name__`` from ``functools.partial`` in ``jax_tracer.py``.

**Related GitHub Issues:** #815 
[sc-66088]

### CODE REVIEW

**Comment by erick-xanadu** [comment]
@tzunghanjuang don't forget to add this to the changelog.

**Review by erick-xanadu — APPROVED** [review]

File: frontend/catalyst/jax_tracer.py [hunk 1]
```diff
@@ -151,7 +151,10 @@ class Function:
     @debug_logger_init
     def __init__(self, fn):
         self.fn = fn
-        self.__name__ = fn.__name__
+        if isinstance(fn, partial):
+            self.__name__ = fn.func.__name__
```
Comment (dime10) [inline frontend/catalyst/jax_tracer.py:155]: This is a bit too specific for my taste, although it works. We might want to consider what to do with other callables that don't define a `__name__` attribute :)

Comment (tzunghanjuang) [reply frontend/catalyst/jax_tracer.py:155]: Maybe always using ``repr`` to get a printable string of the given function?

Comment (josh146) [reply frontend/catalyst/jax_tracer.py:155]: During QA this week, I've run into a couple of situation where I've gotten the error

```python
AttributeError: '?' object has no attribute '__name__'
```

I can't remember exactly what the ? object was though, or how I managed to trigger it :)

**Pull request merged**